lib/index.js

```diff
@@ -111,35 +129,70 @@ const downloadFromInfoCallback = (stream, info, options) => {
       });
     }
 
-    req = miniget(format.url, requestOptions);
-
-    req.on('response', res => {
-      if (stream._isDestroyed) { return; }
-      if (!contentLength) {
-        contentLength = parseInt(res.headers['content-length'], 10);
-      }
-    });
-    req.on('data', ondata);
+    const shouldBeChunked = !format.audioBitrate || !format.qualityLabel;
+    let start = (options.range && options.range.start) || 0;
+    let end = start + dlChunkSize;
+
+    if (shouldBeChunked) {
+      const heads = miniget(format.url);
```

Comment:
most formats now contain a `format.contentLength`, we don't need to make an extra request. below, I did check for the headers, just in case, and since the request was already made. but if we don't have to, let's avoid it.

the formats which don't have this property are segmented formats